draco/analysis/flagging.py

```diff
@@ -3144,6 +3144,176 @@ def process(self, rfimask):
+
+        # Determine valid time range in the mask
+        start = np.abs(mask_time - self.target_time[0]).argmin()
+        end = np.abs(mask_time - self.target_time[-1]).argmin() + 1
+        if start == end:
+            raise ValueError(
+                "No target time values fall within the valid mask time range."
+            )
+
+        # Determine valid mask time window
+        valid_mask_range = slice(start, end)
+        mask_time_valid = mask_time[valid_mask_range]
+
+        # Nearest-interpolation: find the nearest index in target time for each input mask time
+        nearest_indices = np.abs(self.target_time[:, None] - mask_time_valid).argmin(
+            axis=0
+        )
+
+        # The time axis of the adjusted mask
+        valid_target_range = slice(nearest_indices[0], nearest_indices[-1] + 1)
+        new_time = self.target_time[valid_target_range]
+        nnew_time = len(new_time)
+
+        # Determine if conservative spreading can be skipped
+        mapped_times = mask_time[valid_mask_range]
+        if np.array_equal(mapped_times, new_time):
+            self.spread_size = 0
+
+        # Build index arrays for broadcasting
+        freq_idx = np.arange(nfreq_local)[:, None, None]
+        el_idx = np.arange(nel)[None, :, None]
+        new_time_idx = nearest_indices[None, None, :] - nearest_indices[0]
+
+        # Initialize and store adjusted mask
+        mask_adj = np.full((nfreq_local, nel, nnew_time), False)
+        np.logical_or.at(
+            mask_adj, (freq_idx, el_idx, new_time_idx), mask[:, :, valid_mask_range]
+        )
+
+        # Spread the mask conservatively across time
+        for _ in range(self.spread_size):
+            mask_adj[:, :, :-1] |= mask_adj[:, :, 1:]
+            mask_adj[:, :, 1:] |= mask_adj[:, :, :-1]
+
+        # Create output container
+        if isinstance(rfimask, containers.LocalizedRFIMask):
+            out = containers.LocalizedRFIMask(freq=freq, time=new_time, el=el)
+            out.mask.local_data[...] = mask_adj
+        elif rfimask.mask.distributed:
+            out = containers.RFIMask(freq=freq, time=new_time)
+            out.mask.local_data[...] = mask_adj[:, 0, :]
+        else:
+            out = containers.RFIMask(freq=freq, time=new_time)
+            out.mask[:] = mask_adj[:, 0, :]
+
+        # Handle frac_rfi if present
+        if rfimask.datasets.get("frac_rfi", None) is not None:
+
+            # Extract frac_rfi data
+            frac_rfi = (
+                rfimask.frac_rfi.local_data
+                if rfimask.mask.distributed
+                else rfimask.frac_rfi[:]
+            )
+            frac_rfi = frac_rfi.reshape(nfreq_local, nel, ntime)
+
+            # Initialize and store adjusted mask
+            frac_rfi_adj = np.full((nfreq_local, nel, nnew_time), 0.0)
+            np.maximum.at(
+                frac_rfi_adj,
+                (freq_idx, el_idx, new_time_idx),
+                frac_rfi[:, :, valid_mask_range],
+            )
```

Comment:
```suggestion
    np.maximum(frac_rfi_adj, frac_rfi[..., valid_mask_range], out=frac_rfi_adj)
```

Similar to above - since no selections are being made on the first two axes, using `.at` is a bit overkill and confusing to read. This suggestions should accomplish the same thing.